rskj-core/src/main/java/org/ethereum/vm/PrecompiledContracts.java

```diff
@@ -218,6 +224,17 @@ public abstract static class PrecompiledContract {
         public void init(Transaction tx, Block executionBlock, Repository repository, BlockStore blockStore, ReceiptStore receiptStore, List<LogInfo> logs) {
         }
 
+        public void init(ProgramInvoke programInvoke) {
```

Comment:
do we really need two init() methods in this abstract class? You could override the one below in `Environment`

Reply:
We could do that, but it would make the code a little less readable and flexible in case we want to add a similar contract.

Reply:
could you please elaborate a bit more on how this adds flexible? I was just thinking about re-using the approach used for other precompiles..

Reply:
Sure, it is more of a readability issue, imagine we add another contract that requires a different set of arguments, I think it cleaner and easy to read to just add a new init method with just the new arguments. This will also allow us to call the new init from classes that might not have the rest of the old arguments available by just passing the required arguments instead of also passing a lot of null values for the arguments that are not needed, we could also use the init as a factory method manage just the initialization data of an specific precompiled contract, you can think of it as a factory pattern for each init method.

One option would be to group all arguments in a class and use said class as the argument, that way we only need one init method.

Reply:
I like more the idea of args grouping, but this will require quite a good portion of refactoring I suppose. In fact, precompiled contracts are not being added that often.